write.go

```diff
@@ -603,7 +604,12 @@ func writePixelData(w dicomio.Writer, t tag.Tag, value Value, vr string, vl uint
 				}
 			}
 		}
-		if err := w.WriteBytes(buf.Bytes()); err != nil {
+		// if the byte length is even, append 1 padding byte
+		rawData := buf.Bytes()
+		if len(rawData)%2 != 0 {
+			rawData = append(rawData, 0)
+		}
+		if err := w.WriteBytes(rawData); err != nil {
```

Comment:
Not a big deal, but can consider using buf.Len() and only calling buf.Bytes() when the buffer is ready to be written.

```suggestion
		// if the byte length is even, append 1 padding byte
		if buf.Len() % 2 != 0 {
			if err := buf.WriteByte(0); err != nil {
				return err
			}
		}
		if err := w.WriteBytes(buf.Bytes()); err != nil {
```